lab-nicholasc/src/index.js

```diff
@@ -0,0 +1,57 @@
+import './style/main.scss';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import SearchForm from './components/search-form';
+import ResultList from './components/result-list';
+import superagent from 'superagent';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      topics: [],
+      hasSearched: false,
+      loading: false,
+      failedSearch: false,
+    };
+
+    this.search = this.search.bind(this);
+    this.setResults = this.setResults.bind(this);
+  }
+
+  search(subreddit = 'trees', limit = 20) {
```

Comment:
this Reddit API call should be inside of your SearchForm component 

onSubmit the form should make a request to reddit
    - it should make a get request to http://reddit.com/r/${searchFormBoard}.json?limit=${searchFormLimit}
    - on success it should pass the results to the application state
    - on failure it should add a class to the form called error and turn the form's inputs borders red